x/ccv/provider/keeper/msg_server.go

```diff
@@ -139,19 +139,30 @@ func (k msgServer) SubmitConsumerDoubleVoting(goCtx context.Context, msg *types.
 func (k msgServer) OptIn(goCtx context.Context, msg *types.MsgOptIn) (*types.MsgOptInResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	valAddress, err := sdk.ConsAddressFromBech32(msg.ProviderAddr)
+	providerValidatorAddr, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
```

Comment:
Fixed this. Similarly to `AssignConsumerKey` we consider that `ProviderAddr` in the message is the operator address and use this operator address to get the consensus address that is used later on.
We made a similar change in `OptOut`.